tealium/collectors/momentsapi/MomentsAPIModule.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  MomentsAPIModule.swift
+//  tealium-swift
+//
+//  Copyright © 2024 Tealium, Inc. All rights reserved.
+//
+
+import Foundation
+#if momentsapi
+import TealiumCore
+#endif
+
+public class TealiumMomentsAPIModule: Collector {
+    public let id: String = ModuleNames.momentsapi
+    public var config: TealiumConfig
+    public var data: [String: Any]?
+    var diskStorage: TealiumDiskStorageProtocol!
+    var momentsAPI: MomentsAPI?
```

Comment:
Are config and diskStorage needed? 

Can we make the momentsAPI constant and non optional? 
Other modules have the same problem with testing too, but we solved it nicely I think in the `TagManagementModule` and we could do the same here.

Reply:
MomentsAPI has to be optional, because someone may forget to pass in a region in the config. If this happens, we log an error, and don't initialize MomentsAPI.

Reply:
Cool, I missed that.
Might still be worth making it constant and using the same pattern we used in `TagManagementModule` to avoid creating the MomentsAPI in tests just to replace it with the mock.